salt/modules/nacl.py

```diff
@@ -160,6 +171,26 @@ def _get_key(rstrip_newline=True, **kwargs):
     return key
 
 
+def _get_key_pub(rstrip_newline=True, **kwargs):
+    '''
+    Return key pub
+    '''
+    config = _get_config(**kwargs)
+    key_pub = config['key_pub']
+    keyfile_pub = config['keyfile_pub']
+    if not key_pub and keyfile_pub:
+        if not os.path.isfile(keyfile_pub):
+            raise Exception('file not found: {0}'.format(keyfile_pub))
```

Comment:
Why not just catch an exception if the `fopen` fails to avoid the stat overhead every time?